web/pdf_thumbnail_view.js

```diff
@@ -224,8 +202,10 @@ class PDFThumbnailView {
       this.maxCanvasPixels,
       this.maxCanvasDim
     );
-    canvas.width = (width * outputScale.sx) | 0;
-    canvas.height = (height * outputScale.sy) | 0;
+    const canvas = new OffscreenCanvas(
```

Comment:
In the preview it looks like thumbnail generation for e.g. the Tracemonkey PDF is a bit slower than before.

Reply:
That's true:
https://bugzilla.mozilla.org/show_bug.cgi?id=2003060
I'll update the patch to use HTMLCanvasElement.